lib/puppet/node.rb

```diff
@@ -27,7 +27,10 @@ def initialize_from_hash(data)
     @name       = data['name']       || (raise ArgumentError, _("No name provided in serialized data"))
     @classes    = data['classes']    || []
     @parameters = data['parameters'] || {}
-    @environment_name = data['environment']
+    env_name = data['environment']
+    env_name = env_name.intern unless env_name.nil?
+    @environment_name = env_name
+    environment = env_name
```

Comment:
It looks like we already fail both resource and apply when the environment directory doesn't exist because of https://github.com/puppetlabs/puppet/blob/master/lib/puppet/application_support.rb#L29 (we're in `run_mode` user)